pkg/apis/clusteroperator/validation/clusterversion.go

```diff
@@ -69,6 +69,16 @@ func ValidateClusterVersionSpec(spec *clusteroperator.ClusterVersionSpec, fldPat
 		allErrs = append(allErrs, field.Required(vmImagesPath, "must define VM images"))
 	}
 
+	deploymentTypePath := fldPath.Child("deploymentType")
+	if spec.DeploymentType == "" {
+		allErrs = append(allErrs, field.Required(deploymentTypePath, "must define deployment type"))
```

Comment:
I would like for this validation to be stronger than just empty or not empty. We should validate that the deployment type is one of the valid enumerated values.

Reply:
My mistake I thought the use of ClusterDeploymentType for that handled that for us like a java enum, will get this fixed.

Reply:
When in doubt, assume that go doesn't handle anything for us.  ;)